WIP: Ignore: Testing int4 branch

bobvawter opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/32419)
<!-- Reviewable:end -->

bobvawter committed:
sql: Change INT type to mean INT8

This change is the first in a series of changes to support #26925, to better
align Cockroach's integer types with PostgreSQL.

Currently, INT is a 64-bit type, wwhile in PostgreSQL INT is a 32-bit
type. This change adds a cluster and a session setting which control how INT is
interpreted.  The default behavior is to interpret INT as INT8 when used in a
SQL statement, which is consistent with existing behavior.

Release note: None

knz reviewed (commented):
(please ignore this review - these are notes to myself)

> Reviewed 60 of 71 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/32419#-:-LSKjonl7o_SzNnjar63:b-pyev3s)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/sql/logictest/testdata/logic_test/show_source, line 33 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/32419#-LSKgopG1lKl4IEPffnW:-LSKgopG1lKl4IEPffnX:b-qivht0) ([raw file](https://github.com/cockroachdb/cockroach/blob/f577cfd77a7dec9102178c7277a3206da96559d6/pkg/sql/logictest/testdata/logic_test/show_source#L33)):*
> ```
> database                           test
> datestyle                          ISO, MDY
> default_int_size                   INT8
> ```

will be easier for clients to use if this was an integer

if you prefer the _value_ to be a type, then the name of the session var is incorrect (the word "size" implies a number). The name would then become `default_int_normalization`.

I also recommend this be named `experimental_...` until further notice.
___
*[pkg/sql/sem/tree/datum.go, line 1679 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/32419#-LSKgXxy7ySJpLk7o7qP:-LSKgXxy7ySJpLk7o7qQ:b-26kdyh) ([raw file](https://github.com/cockroachdb/cockroach/blob/f577cfd77a7dec9102178c7277a3206da96559d6/pkg/sql/sem/tree/datum.go#L1679)):*
> ```Go
> // the given values.
> func NewParseTimeContext(
> 	mode duration.AdditionMode, intSize intsize.IntSize, relativeParseTime time.Time,
> ```

I do not understand why the integer size is relevant when parsing times.
___
*[pkg/sql/sqlbase/column_type_properties.go, line 679 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/32419#-LSKg5VhBEyuRtRtwHyo:-LSKg5VhBEyuRtRtwHyp:bo63jyi) ([raw file](https://github.com/cockroachdb/cockroach/blob/f577cfd77a7dec9102178c7277a3206da96559d6/pkg/sql/sqlbase/column_type_properties.go#L679)):*
> ```Go
> 	case ColumnType_INT:
> 		if v, ok := tree.AsDInt(inVal); ok {
> 			if size, ok := intsize.FromWidth(int(typ.Width), ctx.GetDefaultIntSize()); ok {
> ```

This needs an explanatory comment to illustrate what happens in the various possible combinations.


<!-- Sent from Reviewable.io -->

knz commented:
@bobvawter I think this PR will not go forward, right?

bobvawter commented:
Dropping.
